src/ensembl/src/content/app/entity-viewer/types/gene.ts

```diff
@@ -37,5 +39,15 @@ export type Gene = {
   so_term: string;
   slice: Slice;
   transcripts: Transcript[];
-  // external_references
+  alternative_symbols: string[];
+  attributes: string[];
+  function?: {
+    description: string;
+    source: Source;
+  };
+  filters?: {
+    [key: string]: JSONValue;
+  };
+  metadata?: Metadata;
+  external_references?: ExternalReference[];
```

Comment:
Several things.

One is that we need to up our types game. On the one hand, we want to describe the types of entities that we can request from Thoas. On the other hand, since every field coming from Thoas is optional, we'll probably need multiple types — one type listing all the fields as potentially undefined, and also narrowed-down types that remove the optionality from the fields that have actually been requested. This is going to be painful and is definitely for another PR.

Another is that among the fields that you added here, the gene will only have metadata and external_references. Alternative symbols, function, etc. will be fields inside the metadata field. We are still working on the shape of metadata items — see https://github.com/Ensembl/ensembl-cdm-docs/pull/18. I don't expect it to be finished before mid-January; and then the data will have to be added to Thoas. So let's not attempt to add these fields to our code prematurely — they won't be ready for at least a couple of months.

The `filters` field does not exist anywhere on the Gene type at all.

I suppose the only fields that we can now show dynamically for the overview panel are the ones I marked with the green checkmark here:

<img src="https://user-images.githubusercontent.com/6834224/102230667-12df7780-3ee5-11eb-88dc-7ea9e8d6b8ff.png" width="300">

@ens-st3 are you expecting anything more from this task?

Reply:
I have removed the UI sections and the thoas data properties in `GeneOverview.ts`